### PR TITLE
New composition bfdr

### PR DESCRIPTION
**Pull request by rpassas**
### Summary

Pulled in updated BFDR repository profiles/examples, created examples for compositions and migrated profiles, and resolved errors created by the profile/example changes.

Per our conversations the following were relatively untouched: inline examples (and bundles as a result), race/ethnicity composition. Additionally, references to patient were not updated (still refer to VRCL) to minimize movement of content.

Composition profiles/bundle do not have a rule set implemented for entries (like in VRDR).

Did not update transition table - saving that for when content moves.

Other items: 
- **plurality** It's included as an edit flag in jurisdiction compositions and is still referenced in inline examples. It was replaced by numberLiveBirthsThisDelivery + numberFetalDeathsThisDelivery
- the profiles that moved to BFDR from VRCL need examples with URLs in their meta.profile

### CODE REVIEW

File: input/fsh/BFDR/bundlescompositions/BundleDocumentBFDRNew.fsh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Profile: BundleDocumentBFDRNew
+Parent: Bundle
+Title: "Bundle - Document Birth and Fetal Death"
+Description: "This Bundle profile represents a Birth and Fetal Death Document Bundle. It contains any one of the Birth and Fetal Death Compositions."
+* ^meta.versionId = "5"
+* ^meta.lastUpdated = "2023-04-25T05:58:33.215+00:00"
+* ^meta.source = "#9Kkc7086j0eYQVSm"
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/BundleDocumentBFDRNew.fsh:7]: Delete this trifolia detritus wherever found.

File: input/fsh/BFDR/bundlescompositions/BundleDocumentBFDRNew.fsh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+* ^experimental = false
+* ^publisher = "HL7 Public Health Work Group"
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/BundleDocumentBFDRNew.fsh:9]: Delete ^publisher wherever found.

File: input/fsh/BFDR/bundlescompositions/CompositionCodedCauseOfFetalDeathNew.fsh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Parent: Composition
+Title: "Composition - Coded Cause of Fetal Death"
+Description: "This profile communicates coded cause of fetal death information to appropriate jurisdictional Vital Records Offices."
+* ^meta.versionId = "25"
+* ^meta.lastUpdated = "2023-04-25T05:58:06.375+00:00"
+* ^meta.source = "#CHyKpgkzkwbhHXuJ"
+* ^status = #active
+* ^experimental = false
+* ^publisher = "HL7 Public Health Work Group"
+* ^jurisdiction = urn:iso:std:iso:3166#US "United States of America"
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionCodedCauseOfFetalDeathNew.fsh:11]: Delete wherever found.

File: input/fsh/BFDR/bundlescompositions/CompositionJurisdictionFetalDeathReportNew.fsh [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+  * code 1.. 
+  * code = $loinc#55752-0
+    * ^short = "Clinical information"
+    * ^definition = "Clinical information"
+  * focus 1.. 
+  * focus only Reference(PatientMotherVitalRecords)
+    * ^short = "The focus of this section is the mother"
+    * ^definition = "The subject of the composition as a whole is the fetus baby. The focus of this section in the composition is the mother."
+  * entry 1.. 
+    * ^slicing.discriminator.type = #profile
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionJurisdictionFetalDeathReportNew.fsh:129]: Make/find a ruleset for profile-based-open-slicing and replace everywhere.

File: input/fsh/BFDR/bundlescompositions/CompositionJurisdictionFetalDeathReportNew.fsh [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    * ^definition = "The subject of the composition as a whole is the fetus baby. The focus of this section in the composition is the father."
+* section[editFlags] ^short = "Edit Flags Section"
+  * ^definition = "This section contains the edit flags that specify whether the originally provided data passed validation checks. These validation checks are done at the jurisdiction prior to sending to the National Statistical Agency and are based on edit criteria in the Birth Edit Specifications for the 2003 Proposed Revision of the U.S. Standard Certificate of Birth and the Fetal Death Edit Specifications for the 2003 Revision of the U.S. Standard Report of Fetal Death."
+  * code 1.. 
+  * code = $loinc#92012-4
+    * ^short = "Edit flags section"
+  * entry ..8 
+    * ^slicing.discriminator.type = #profile
+    * ^slicing.discriminator.path = "$this.resolve()"
+    * ^slicing.rules = #open
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionJurisdictionFetalDeathReportNew.fsh:255]: This should also include unordered (probably the default), so can use the to-be-found/defined ruleset

File: input/fsh/BFDR/bundlescompositions/CompositionJurisdictionLiveBirthReportNew.fsh [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+Parent: Composition
+Title: "Composition - Jurisdiction Live Birth Report"
+Description: "This Composition profile contains information of a live birth and the issuance of a Birth Certificate to be recorded and communicated to the national statistics agency."
+* ^meta.versionId = "109"
+* ^meta.lastUpdated = "2023-04-25T05:58:09.079+00:00"
+* ^meta.source = "#POHh5pjMTdqg9crJ"
+* ^status = #active
+* ^experimental = false
+* ^publisher = "HL7 Public Health Work Group"
+* ^jurisdiction = urn:iso:std:iso:3166#US "United States of America"
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionJurisdictionLiveBirthReportNew.fsh:11]: delete

File: input/fsh/BFDR/bundlescompositions/CompositionJurisdictionLiveBirthReportNew.fsh [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+* . ^short = "Jurisdiction Live Birth Report"
+* extension 1.. 
+  * ^slicing.discriminator.type = #value
+  * ^slicing.discriminator.path = "url"
+  * ^slicing.rules = #open
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionJurisdictionLiveBirthReportNew.fsh:16]: I think this is the default for extension slices.  Can probably be deleted everywhere.

File: input/fsh/BFDR/bundlescompositions/CompositionJurisdictionLiveBirthReportNew.fsh [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+  * focus 1.. 
+  * focus only Reference(PatientMotherVitalRecords)
+    * ^short = "The focus of this section is the mother"
+    * ^definition = "The subject of the composition as a whole is the newborn baby. The focus of this section in the composition is the mother."
+  * entry 1.. 
+    * ^slicing.discriminator.type = #profile
+    * ^slicing.discriminator.path = "$this.resolve()"
+    * ^slicing.ordered = false
+    * ^slicing.rules = #open
+    * ^short = "Entries that are contained in the Mother Prenatal section"
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionJurisdictionLiveBirthReportNew.fsh:77]: profile-based open slicing, unordered --> ruleset.

File: input/fsh/BFDR/bundlescompositions/CompositionProviderFetalDeathReportNew.fsh [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+Parent: Composition
+Title: "Composition - Provider Fetal Death Report"
+Description: "This Composition profile contains constraints to address the use case describing the need for fetal death information to be recorded and communicated to the jurisdictional Vital Records Office."
+* ^meta.versionId = "69"
+* ^meta.lastUpdated = "2023-04-25T05:58:09.576+00:00"
+* ^meta.source = "#q3kYsclRE3ANIm1f"
+* ^status = #active
+* ^experimental = false
+* ^publisher = "HL7 Public Health Work Group"
+* ^jurisdiction = urn:iso:std:iso:3166#US "United States of America"
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionProviderFetalDeathReportNew.fsh:11]: delete

File: input/fsh/BFDR/bundlescompositions/CompositionProviderFetalDeathReportNew.fsh [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+  * ^short = "The Encounter for the Composition is the mother's maternity encounter."
+  * ^definition = "The Encounter for the Composition is the mother's maternity encounter."
+* author ..1 
+* author only Reference(PractitionerVitalRecordsNew)
+  * ^short = "The author is the person who verifies/approves the accuracy of the data to be sent to the vital records system."
+  * ^definition = "Name and title of person completing the report. May be, but need not be, the same as the attendant at delivery."
+* section 1.. 
+  * ^slicing.discriminator.type = #pattern
+  * ^slicing.discriminator.path = "code"
+  * ^slicing.rules = #open
```
Comment (saulakravitz) [inline input/fsh/BFDR/bundlescompositions/CompositionProviderFetalDeathReportNew.fsh:47]: I think this is the default for composition sections.   May be able to delete this.   Everywhere there are composition sections.

File: input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-10.fsh
Comment (saulakravitz) [inline input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-10.fsh:1]: I thought we were going to get rid of all of the inline instances, a la VRDR.
They aren't shown as examples of individual profiles, which is a shame.
All of the examples I looked at look they been repurposed from one type of observation to another which makes me scratch my head.

File: input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-16.fsh
Comment (saulakravitz) [inline input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-16.fsh:1]: THis doesn't look right.   Switching out smoking for education....

File: input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-14.fsh
Comment (saulakravitz) [inline input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-14.fsh:1]: THis doesn't look right... Switching out plurality for WIC

File: input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-13.fsh
Comment (saulakravitz) [inline input/fsh/BFDR/instances/Inline-Instance-for-bundle-jurisdiction-fetal-death-not-named-13.fsh:1]: THis doesn't look right.  Switching out normal delivery finding for prepregnancy weight.

File: input/fsh/BFDR/migratingInstances/observation-apgar-score-babyg-quinn-1-min-new.fsh [hunk 11]
```diff
@@ -0,0 +1,14 @@
+Instance: observation-apgar-score-babyg-quinn-1-min-new
+InstanceOf: ObservationApgarScoreNew
+Title: "Observation - Apgar Score - BabyG Quinn: 1 min"
+Description: "Observation - Apgar Score: BabyG Quinn: 1 min example"
+Usage: #example
+* meta
+  * versionId = "4"
+  * lastUpdated = "2023-04-06T00:25:56.510Z"
```
Comment (saulakravitz) [inline input/fsh/BFDR/migratingInstances/observation-apgar-score-babyg-quinn-1-min-new.fsh:8]: delete -- in all instances.

File: input/fsh/BFDR/valuesets/ReplaceStatus.fsh [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+ValueSet: ValueSetReplacementStatus
+Id: ValueSet-replacement-status
+Title: "ValueSet - Replacement Status"
+Description: "This ValueSet contains codes that represent the replacement status of a record submission."
+* ^meta.versionId = "1"
+* ^meta.lastUpdated = "2023-06-21T23:24:24.587Z"
+* ^url = "http://hl7.org/fhir/us/bfdr/ValueSet/ValueSet-replacement-status"
+* ^version = "1.1.0"
+* ^status = #active
```
Comment (saulakravitz) [inline input/fsh/BFDR/valuesets/ReplaceStatus.fsh:13]: delete

File: input/fsh/BFDR/valuesets/ReplaceStatus.fsh [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+* ^experimental = false
+* ^publisher = "HL7 International - Public Health Work Group"
```
Comment (saulakravitz) [inline input/fsh/BFDR/valuesets/ReplaceStatus.fsh:15]: delete

**Review by saulakravitz — CHANGES_REQUESTED** [review]

File: input/fsh/BFDR/bundlescompositions/BundleDocumentBFDRNew.fsh [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  * ^short = "BFDR Document Bundle identifier"
+* type = #document (exactly)
+* type 
+* entry 
+  * ^slicing.discriminator.type = #profile
+  * ^slicing.discriminator.path = "$this.resource"
+  * ^slicing.rules = #open
+* entry contains BFDRComposition 1..1 
+* entry[BFDRComposition].resource 1.. 
+* entry[BFDRComposition].resource only Composition
```
Comment (ptran265) [inline input/fsh/BFDR/bundlescompositions/BundleDocumentBFDRNew.fsh:15]: Not sure if generalization is intentional here, but the original bundle references the 4 specific compositions: 

* resource only CompositionProviderLiveBirthReport or CompositionProviderFetalDeathReport or CompositionJurisdictionLiveBirthReport or CompositionJurisdictionFetalDeathReport

File: input/fsh/BFDR/bundlescompositions/CompositionProviderLiveBirthReportNew.fsh [hunk 15]
```diff
@@ -0,0 +1,296 @@
...
+  * entry contains
+      pregnancyRiskFactors 0..* and
+      numberPreviousCesareans 0..1  and
+      infectionsDuringPregnancy 0..* and
+      obstetricProcedures 1..1  and
+      characteristicsLaborDelivery 0..* and
+      fetalPresentation 0..1  and
+      finalRouteMethodDelivery 0..1 and
+      maternalMorbidity 0..*
+  * entry[pregnancyRiskFactors] only Reference(Condition-prepregnancy-diabetes-vr or Condition-gestational-diabetes-vr or Condition-prepregnancy-hypertension-vr or Condition-gestational-hypertension-vr or Condition-eclampsia-hypertension-vr or Observation-previous-preterm-birth-vr or Procedure-infertility-treatment-vr or Procedure-artificial-insemination-vr or Procedure-assisted-fertilization-vr or Observation-previous-cesarean-vr or Observation-none-of-specified-pregnancy-risk-factors-vr)
```
[the remark this answers is not in the document]
Comment (rpassas) [reply input/fsh/BFDR/bundlescompositions/CompositionProviderLiveBirthReportNew.fsh:161]: I assume you mean reference by ID, not profile. I think this was a goFSH conversion - changing it to profile references.

**Review by saulakravitz — APPROVED** [review]
Let's merge and move forward.

**Pull request merged**